x/leveragelp/keeper/msg_server_close_positions.go

```diff
@@ -26,37 +24,21 @@ func (k msgServer) ClosePositions(goCtx context.Context, msg *types.MsgClosePosi
 		if !poolFound {
 			continue
 		}
-		ammPool, poolErr := k.GetAmmPool(ctx, position.AmmPoolId)
-		if poolErr != nil {
-			continue
-		}
 
 		cachedCtx, write := ctx.CacheContext()
-		_, _, _, err = k.CheckAndLiquidateUnhealthyPosition(cachedCtx, &position, pool, ammPool)
+		_, _, _, err = k.CheckAndLiquidateUnhealthyPosition(cachedCtx, &position, pool)
 		if err == nil {
 			write()
+			liqLog = append(liqLog, position.Id)
 		}
 		if err != nil {
-			// Add log about error or not liquidated
-			liqLog = append(liqLog, fmt.Sprintf("Position: Address:%s Id:%d cannot be liquidated due to err: %s", position.Address, position.Id, err.Error()))
-		}
-
-		if k.hooks != nil {
-			// ammPool will have updated values for opening position
-			found := false
-			ammPool, found = k.amm.GetPool(ctx, position.AmmPoolId)
-			if !found {
-				return nil, errorsmod.Wrap(types.ErrPoolDoesNotExist, fmt.Sprintf("poolId: %d", position.AmmPoolId))
-			}
-			err = k.hooks.AfterLeverageLpPositionClose(ctx, position.GetOwnerAddress(), ammPool)
-			if err != nil {
-				return nil, err
-			}
+			liqLog = append(liqLog, position.Id)
+			ctx.Logger().Error(fmt.Sprintf("Unhealthy Position: Address:%s Id:%d cannot be liquidated due to err: %s", position.Address, position.Id, err.Error()))
 		}
 	}
 
 	// Handle stop loss
-	closeLog := []string{}
+	closeLog := []uint64{}
```

Comment:
we won't be able to see the reason for not liquidation if we change it to uint64

Reply:
Logging the reason in even would be bad as it would take lot of space, it will be there in logger though. We will be able to see it, just won't be easy. But what are chances of this happening.